internal/publisher/publiser.go

```diff
@@ -0,0 +1,15 @@
+package publisher
+
+import (
+	"context"
+
+	"github.com/KyberNetwork/evmlistener/pkg/types"
+)
+
+type Client interface {
```

Comment:
I think we can remove this interface.

Reply:
I think this interface helps us with flexibility.
Since we can have the same bizz message logic, but a different message queue.
You can think PublishService is a service layer, and Client is a repository layer (in the repository coding pattern).
So a service can switch its repo easily.

Reply:
No need to do that, the publisher already tells which backend it uses.

Reply:
Also, I don't want to keeper that interface because we have to add `extra` mapping for it. Not every message queue will need that information.